3p/ampcontext.js

```diff
@@ -62,24 +120,33 @@ export class AmpContext {
    *    every time we receive a page visibility message.
    */
   observePageVisibility(callback) {
-    return this.client_.makeRequest(
-        MessageType.SEND_EMBED_STATE,
-        MessageType.EMBED_STATE,
-        callback);
+    return this.visibilityObservable_.add(callback);
   };
 
   /**
    *  Send message to runtime to start sending intersection messages.
-   *  @param {function(Object)} callback Function to call every time we receive
-   *  an intersection message.
+   *  @param {function(Array<Object>)} callback Function to call every time we
+   *    receive an intersection message.
    *  @returns {function()} that when called stops triggering the callback
    *    every time we receive an intersection message.
    */
   observeIntersection(callback) {
-    return this.client_.makeRequest(
+    const unlisten = this.client_.makeRequest(
```

Comment:
should switch to observable as well, so that observeIntersection can be called multiple times.